options.go

```diff
@@ -10,6 +29,9 @@ type Options struct {
 	// this, an error is returned when loading the tree. Only used for the initial SaveVersion()
 	// call.
 	InitialVersion uint64
+
+	// When Stat is not nil, statistical logic needs to be executed
+	Stat *Statistics
```

Comment:
Is it possible to define this on `nodeDb` instead?

Options are used for configuring iavl. On the other hand, this seems non-configuration specific

Reply:
https://github.com/facebook/rocksdb/blob/main/include/rocksdb/options.h#L609
Usage is just like rocksdb, eg:
```
Options options;
options.statistics = ROCKSDB_NAMESPACE::CreateDBStatistics();
Status s = DB::Open(options, kDBPath, &db);
...
options.statistics->getTickerCount(NUMBER_BLOCK_COMPRESSED);
```
Our usage:
```
opt.Statistics = MyStatistics
NewMutableTreeWithOpts(db, 100, &opt);
...
print(MyStatistics)
```

Reply:
Thanks for the example.

Although it might be convenient in C++, I don't think it would be as pretty in Go

Relying on `opts.Statistics` pointer is prone to data races. In the linked example, the synchronization of the counters is likely handled in `getTickerCount(...)`

In our case, we currently don't have such a method. However, even if we had one, relying on the user on calling that method instead of accessing `opt.Statistics` directly is dangerous. Someone could easily misuse it and access one of the counters directly without a lock (e.g. `opts.Statistics.CacheHitCnt`) causing a data race.

So I think we could do one of the following:
- avoid exporting `CacheHitCnt` and `CacheMissCnt` and create methods like:
```
func (s *Statistics) GetCacheHitCnt() {...}
```
in which we atomically get the counters

OR

- define the Statistics on `nodeDb` instead of `Options`, expose getters on the mutable tree and atomically get the counters

Please let me know what you think

Reply:
I prefer expose getters on the Statistics than mutable tree, because more statistics item may be added in the future, and their getters will be more concise to be hidden on the Statistics.
And I also prefer to add Statistics to option, because statistical behavior could cause performance damage and no every one can accept this. So it is better to be optional.

Reply:
Good call, sounds good to me